### PR TITLE
impl Clone for all containers

### PR DESCRIPTION
**Pull request by vorot93**

### CODE REVIEW

File: src/indexset.rs [hunk 1]
```diff
@@ -44,6 +44,7 @@ pub type FnvIndexSet<T, N> = IndexSet<T, N, BuildHasherDefault<FnvHasher>>;
 ///     println!("{}", book);
 /// }
 /// ```
+#[derive(Clone)]
```
Comment (japaric) [inline src/indexset.rs]: This should be a manual `Clone` implementation. No need for a `Clone` bound on `N`.

File: src/linear_map.rs [hunk 2]
```diff
@@ -8,6 +8,7 @@ use Vec;
 /// A fixed capacity map / dictionary that performs lookups via linear search
 ///
 /// Note that as this map doesn't use hashing so most operations are **O(N)** instead of O(1)
+#[derive(Clone)]
```
Comment (japaric) [inline src/linear_map.rs]: This should be a manual `Clone` implementation. No need for a `Clone` bound on `N`.

File: src/vec.rs [hunk 3]
```diff
@@ -336,6 +348,7 @@ where
 ///
 /// [`Vec`]: (https://doc.rust-lang.org/std/vec/struct.Vec.html)
 ///
+#[derive(Clone)]
```
Comment (japaric) [inline src/vec.rs]: This should be a manual `Clone` implementation. No need for a `Clone` bound on `N`.

File: src/string.rs [hunk 4]
```diff
@@ -6,6 +6,7 @@ use generic_array::ArrayLength;
 use Vec;
 
 /// A fixed capacity [`String`](https://doc.rust-lang.org/std/string/struct.String.html)
+#[derive(Clone)]
```
Comment (japaric) [inline src/string.rs]: This should be a manual `Clone` implementation. No need for a `Clone` bound on `N`.

File: src/spsc/mod.rs [hunk 5]
```diff
@@ -384,6 +384,7 @@ impl_!(u16, u16_sc);
 impl_!(usize, usize_sc);
 
 /// An iterator over the items of a queue
+#[derive(Clone)]
```
Comment (japaric) [inline src/spsc/mod.rs]: This should be a manual `Clone` implementation. No need for `Clone` bounds on any of the type parameters.

File: src/linear_map.rs [hunk 6]
```diff
@@ -403,6 +404,7 @@ where
     }
 }
 
+#[derive(Clone)]
```
Comment (japaric) [inline src/linear_map.rs]: This should be a manual `Clone` implementation. No need for `Clone` bounds on any of the type parameters.

File: src/indexmap.rs [hunk 7]
```diff
@@ -826,6 +829,7 @@ where
     }
 }
 
+#[derive(Clone)]
```
Comment (japaric) [inline src/indexmap.rs]: This should be a manual `Clone` implementation. No need for `Clone` bounds on any of the type parameters.

File: src/indexset.rs [hunk 8]
```diff
@@ -549,6 +550,7 @@ where
     }
 }
 
+#[derive(Clone)]
```
Comment (japaric) [inline src/indexset.rs]: This should be a manual `Clone` implementation. No need for `Clone` bounds on any of the type parameters.

**Comment by vorot93** [comment]
@japaric Replaced derives with manual impls where necessary

**Comment by japaric** [comment]
@vorot93 thanks!

bors r+

**Pull request merged**